### PR TITLE
46650 connect search page with details page

### PR DESCRIPTION
**Pull request by RaphaelFortin-Beslogic**

### CODE REVIEW

File: mono-repo/libs/catalog/feature-details/src/lib/catalog-feature-details.component.ts [hunk 1]
```diff
@@ -19,7 +19,8 @@ export class CatalogFeatureDetailsComponent implements OnInit {
   constructor(private router: Router, private route: ActivatedRoute, private dealFacade: DealFacade) {}
 
   ngOnInit() {
-    this.dealFacade.Get("123TEST123").subscribe((dealDetails) => {
+    const id = this.route.snapshot.paramMap.get('id') ?? '';
```
Comment (RaphaelFortin-Beslogic) [inline mono-repo/libs/catalog/feature-details/src/lib/catalog-feature-details.component.ts:22]: J'etais pas sure ici de la meilleur facon de faire

**Review by RaphaelFortin-Beslogic — COMMENTED** [review]
Jai ajouter 4 differents response.

File: mono-repo/libs/catalog/feature-details/src/lib/catalog-feature-details.component.ts (hunk 1 above)
Comment (helenelebrun-beslogic) [reply mono-repo/libs/catalog/feature-details/src/lib/catalog-feature-details.component.ts:22]: ## 1. When to use route.snapshot.paramMap ?

If you intend not to update your URL parameter within the same component you are accessing it, then you can use the snapshot.

As the name suggests, the parameter would only be accessed once, when the component loads. Hence, it won’t be updated, even if you change its value from within the same component.

```
ngOnInit() {
  this.car = this.route.snapshot.paramMap.get("carBrand")
}
```

## 2. When to use route.paramMap.subscribe ?
If you intend to update the URL parameter within the same component, then you have to use a subscription.

The good news is that it works just like any other subscription in your Angular app. 😎

This technique is only useful if you plan to change the URL parameters within the current route.

```
ngOnInit() {
  this.route.paramMap.subscribe(params => {
    this.car = params.get("carBrand")
  })
}
```

---

Please refer to this article : https://medium.com/@tiboprea/accessing-url-parameters-in-angular-snapshot-vs-subscription-efc4e70f9053

Comment (RaphaelFortin-Beslogic) [reply mono-repo/libs/catalog/feature-details/src/lib/catalog-feature-details.component.ts:22]: oh merci ! :D

**Review by helenelebrun-beslogic — APPROVED** [review]

**Pull request merged**